lib/matplotlib/font_manager.py

```diff
@@ -855,7 +847,6 @@ def set_weight(self, weight):
         except ValueError:
             if weight not in weight_dict:
                 raise ValueError("weight is invalid")
-            weight = weight_dict[weight]
```

Comment:
It seems that "hair" is not a standard weight name, and I do not know a way to map a non-standard weight name to a numeric weight correctly.